odl/discr/discr_mappings.py

```diff
@@ -32,97 +32,88 @@
+                             ''.format(map_type_in))
+        if not isinstance(fspace, FunctionSpace):
+            raise TypeError('`fspace` {!r} is not a `FunctionSpace` '
+                            'instance'.format(fspace))
 
         if not isinstance(partition, RectPartition):
             raise TypeError('`partition` {!r} is not a `RectPartition` '
                             'instance'.format(partition))
-        if not isinstance(dspace, NtuplesBase):
-            raise TypeError('`dspace` {!r} is not an `NtuplesBase` instance'
+        if not isinstance(dspace, TensorSpace):
+            raise TypeError('`dspace` {!r} is not a `TensorSpace` instance'
                             ''.format(dspace))
 
-        if not fset.domain.contains_set(partition):
+        if not fspace.domain.contains_set(partition):
             raise ValueError('{} not contained in the domain {} '
                              'of the function set {}'
-                             ''.format(partition, fset.domain, fset))
+                             ''.format(partition, fspace.domain, fspace))
 
-        if dspace.size != partition.size:
-            raise ValueError('size {} of the data space {} not equal '
-                             'to the size {} of the partition'
-                             ''.format(dspace.size, dspace, partition.size))
+        if dspace.shape != partition.shape:
+            raise ValueError('`dspace.shape` not equal to `partition.shape`: '
+                             '{} != {}'
+                             ''.format(dspace.shape, partition.shape))
 
-        domain = fset if map_type_ == 'sampling' else dspace
-        range = dspace if map_type_ == 'sampling' else fset
+        domain = fspace if map_type == 'sampling' else dspace
+        range = dspace if map_type == 'sampling' else fspace
         Operator.__init__(self, domain, range, linear=linear)
         self.__partition = partition
 
         if self.is_linear:
-            if not isinstance(fset, FunctionSpace):
-                raise TypeError('`fset` {!r} is not a `FunctionSpace` '
-                                'instance'.format(fset))
-            if not isinstance(dspace, FnBase):
-                raise TypeError('`dspace` {!r} is not an `FnBase` instance'
-                                ''.format(dspace))
-            if fset.field != dspace.field:
-                raise ValueError('`field` {} of the function space and `field`'
-                                 ' {} of the data space are not equal'
-                                 ''.format(fset.field, dspace.field))
-
-        order = str(kwargs.pop('order', 'C'))
-        if str(order).upper() not in ('C', 'F'):
-            raise ValueError('`order` {!r} not recognized'.format(order))
-        else:
-            self.__order = str(order).upper()
+            if not fspace.has_field:
+                raise TypeError('`fspace.field` cannot be `None` for '
```

Comment:
Right, maybe it was paranoia that made me hide the `None` behind a property. Will remove it.